### PR TITLE
fix: increase max connections count to support clusters with very large number of CRDs

### PR DESCRIPTION
**Pull request by alexmt**
PR increases the number of connections per host in HTTP transport used by K8S rest config.

### CODE REVIEW

File: pkg/apis/application/v1alpha1/types.go [hunk 1]
```diff
@@ -2172,6 +2175,32 @@ func (proj AppProject) IsDestinationPermitted(dst ApplicationDestination) bool {
 	return false
 }
 
+// SetK8SConfigDefaults sets Kubernetes REST config default settings
+func SetK8SConfigDefaults(config *rest.Config) error {
```
Comment (alexmt) [inline pkg/apis/application/v1alpha1/types.go]: This function should be in `util/kube` package but unfortunately, we have a circular dependency . As soon as `gitops` engine branch is merged this case be fixed.

File: pkg/apis/application/v1alpha1/types.go [hunk 2]
```diff
@@ -2172,6 +2175,32 @@ func (proj AppProject) IsDestinationPermitted(dst ApplicationDestination) bool {
...
+		KeepAlive: 30 * time.Second,
+	}).DialContext
+	// Cache a single transport for these options
+	utilnet.SetTransportDefaults(&http.Transport{
+		Proxy:               http.ProxyFromEnvironment,
+		TLSHandshakeTimeout: 10 * time.Second,
+		TLSClientConfig:     tlsConfig,
+		MaxIdleConns:        500,
+		MaxIdleConnsPerHost: 500,
+		MaxConnsPerHost:     500,
```
Comment (jessesuen) [inline pkg/apis/application/v1alpha1/types.go]: 1. Can you follow the technique which I did for `common.K8sClientConfigQPS` and get this value from some new environment variables, or a default?

2. In the same file, can we document when this value should be tuned, and also remind ourselves what the defaults are if we did not adjust it?

Comment (alexmt) [reply pkg/apis/application/v1alpha1/types.go]: done

File: test/e2e/git_submodule_test.go [hunk 3]
```diff
@@ -11,20 +11,20 @@ import (
...
+//		Recurse().
+//		CustomSSHKnownHostsAdded().
+//		SubmoduleSSHRepoURLAdded(true).
+//		When().
+//		CreateFromFile(func(app *Application) {}).
+//		Sync().
+//		Then().
+//		Expect(SyncStatusIs(SyncStatusCodeSynced)).
+//		Expect(Pod(func(p v1.Pod) bool { return p.Name == "pod-in-submodule" }))
+//}
```
Comment (jessesuen) [inline test/e2e/git_submodule_test.go]: Is this a mistake?

**Review by jessesuen — APPROVED** [review]

File: test/e2e/git_submodule_test.go (hunk 3 above)
Comment (alexmt) [reply test/e2e/git_submodule_test.go]: yes, reverted

**Pull request merged**